python/example_load_gen.py

```diff
@@ -0,0 +1,307 @@
+              success: {context.global_success_count} ({self.percent_requests(context, context.global_success_count)}%) ({self.tps(context, context.global_success_count)} tps)
+          unavailable: {context.global_unavailable_count} ({self.percent_requests(context, context.global_unavailable_count)}%)
+    deadline exceeded: {context.global_deadline_exceeded_count} ({self.percent_requests(context, context.global_deadline_exceeded_count)}%)
+    
+cumulative set latencies:
+{self.output_histogram_summary(context.set_latencies)}
+
+cumulative get latencies:
+{self.output_histogram_summary(context.get_latencies)}
+""")
+
+    async def issue_async_set_get(
+            self,
+            client: simple_cache_client.SimpleCacheClient,
+            context: BasicPythonLoadGenContext,
+            worker_id: int,
+            operation_id: int
+    ) -> None:
+        cache_key = f"worker{worker_id}operation{operation_id}"
+        set_start_time = perf_counter_ns()
+        result: Optional[CacheSetResponse] = await self.execute_request_and_update_context_counts(
+            context,
+            lambda: client.set(self.cache_name, cache_key, self.cache_value)
+        )
+        if result:
+            set_duration = self.get_elapsed_millis(set_start_time)
+            context.set_latencies.record_value(set_duration)
+
+        get_start_time = perf_counter_ns()
+        get_result: Optional[CacheGetResponse] = await self.execute_request_and_update_context_counts(
+            context,
+            lambda: client.get(self.cache_name, cache_key)
+        )
+        if get_result:
+            get_duration = self.get_elapsed_millis(get_start_time)
+            context.get_latencies.record_value(get_duration)
+            if get_result.status() == CacheGetStatus.HIT:
+                value = get_result.value()
+                value_string = f"{value[0:10]}... (len: {len(value)})"
+            else:
+                value_string = 'n/a'
+
+            if context.global_request_count % 1_000 == 0:
+                self.logger.info(
+                    f"worker: {worker_id}, worker request: {operation_id}, global request: {context.global_request_count}, status: {get_result.status()}, val: {value_string}"
+                )
+
+    T = TypeVar("T")
+
+    async def execute_request_and_update_context_counts(
+            self,
+            context: BasicPythonLoadGenContext,
+            block: Callable[[], Coroutine[None, None, T]]
+    ) -> Optional[T]:
+        result, response = await self.execute_request(block)
+        self.update_context_counts_for_request(context, result)
+        return response
+
+    async def execute_request(
+            self,
+            block: Callable[[], Coroutine[None, None, T]]
+    ) -> Tuple[AsyncSetGetResult, Optional[T]]:
+        try:
+            result = await block()
+            return AsyncSetGetResult.SUCCESS, result
+        except InternalServerError as e:
+            # TODO verify exception type
```

Comment:
Right now I'm blindly returning my `UNAVAILABLE` enum value as a result from this function, because this exception doesn't have enough information on it to allow me to check for the actual cause.  When we do our spike on improving error handling in the coming days/weeks this needs to get cleaned up.  Either the exceptions need to provide more info so that we can detect UNAVAILABLE, or I need to change the enums in these load generators to be less specific to particular types of exceptions, etc.